### DISCUSSION TITLE
`BuildTarget` should contain path to config file

### DISCUSSION THREAD

**Discussion by abrams27** [build-server-protocol/build-server-protocol#510] [Proposals]
i believe that it'd be nice to include info about the location of config file(s) in `BuildTarget` - for bazel it'd be the location of the `BUILD` file; it's useful for the client to mark them as build tool files (e.g. to triger auto-reload on change)

another option is a separate endpoint like `buildTarget/configFile`

what do u think?

**Comment by jastice** [comment]
It's a bit complicated. For example in sbt, a project can have its own sbt files and project directory, but it can also be configured from the root sbt files and project directory. On the other hand, config files may not exist at all in simple cases. 

But generally I think it's a good idea to provide this information somehow.

**Comment by abrams27** [comment]
it doesnt have to be a 1-1 mapping; one config file can configure multiple targets. On top of that we can (we should?) also provide project-wide config files (e.g. for bazel ht `WORKSPACE` file)

**Comment by jastice** [comment]
Considering it might be many files, and not every client might be interested in them, I think it's better to put this data in a separate request. What do you think?

**Reply by abrams27** [reply]
yea that works as well, it could return a struct with global configs and a list with target -> config mapping (or the other way around)
